src/sources/core.js

```diff
@@ -29,15 +35,52 @@ class CoreSource extends Source {
     return new CoreDataset(this, pathParts);
   }
 
-  // The computation of these globals should move here.
+  // The computation of these globals should move here. 
   secondTreeOptions(path) {
     return (global.availableDatasets.secondTreeOptions[this.name] || {})[path] || [];
   }
 
+  async collectResources() {
+    if (!this._allResources) this._allResources = new Map();
+    const s3objects = await parseInventory({...this.inventoryLocation, name: this.name});
+    if (!s3objects.length) {
+      // potentially due to a (transient) failure to parse the inventory -- so we want
+      // to keep any previously set resources
+      return;
+    }
+    /* Comment (to drop before merge)
+    This conditional could be replaced by overwriting `collectResources` in CoreStagingSource.
+    That would require most of the logic in this function to be pulled out into a utility function,
+    which may not be a bad idea */
+    const CollectedResources = this.name==='core' ?
+      CoreCollectedResources : CoreStagingCollectedResources;
+    const datasets = new Map(), files = new Map();
+    s3objects.forEach((object) => {
+      const [name, resourceType] = CollectedResources.objectName(object);
+      if (!name) return;
+      if (resourceType === 'dataset') {
+        datasets.has(name) ? datasets.get(name).push(object) : datasets.set(name, [object]);
+      } else if (resourceType === 'file') {
+        files.has(name) ? files.get(name).push(object) : files.set(name, [object]);
+      }
+    })
+    this._allResources.set(
+      'dataset',
+      Array.from(datasets).map(([, objects]) => new CollectedResources(this, objects))
+    );
+    this._allResources.set(
+      'file',
+      Array.from(files).map(([, objects]) => new CollectedResources(this, objects))
+    );
+    // TODO XXX narratives
+  }
```

Comment:
There's an inherent linking between (some) datasets and (some) intermediate files. It would be nice to surface this - e.g. when looking at the card for zika you could see the available intermediate files as well. I ummed and ahhed about trying to build this into this version but ultimately left it as a future todo. Beyond the obvious "how do we programmatically know which datasets are connected to which files", the desired UI needs to be worked out before we know how best to approach it. Note that such a linking is cross-source, so if it's done ahead of time on the server, we'll have to perform a per-linked-resource authz filtering (e.g. the intermediate files for a dataset might be stored in a source you don't have permission to view).